rest/route/host.go

```diff
@@ -16,13 +16,106 @@ import (
 	"github.com/pkg/errors"
 )
 
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/hosts
+
+type hostsChangeStatusesHandler struct {
+	HostToStatus map[string]map[string]string
+	sc           data.Connector
+}
+
+func makeChangeHostsStatuses(sc data.Connector) gimlet.RouteHandler {
+	return &hostsChangeStatusesHandler{
+		sc: sc,
+	}
+}
+
+func (h *hostsChangeStatusesHandler) Factory() gimlet.RouteHandler {
+	return &hostsChangeStatusesHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *hostsChangeStatusesHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	if err := util.ReadJSONInto(body, &h.HostToStatus); err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+
+	if len(h.HostToStatus) == 0 {
+		return fmt.Errorf("Missing host id and status")
+	}
+	for host, json := range h.HostToStatus {
+		if !util.StringSliceContains(evergreen.ValidUserSetStatus, json["status"]) {
```

Comment:
I think json["status"] is a map[string]string, so the StringSliceContains feels wrong.

Reply:
`json["status"]` is a string.